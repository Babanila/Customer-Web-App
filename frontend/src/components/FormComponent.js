import React from 'react'
import { cx, css } from 'emotion'
import { utcDateConverter } from './Utils'

const formStyle = css`
  width: 99%;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  color: #000000;
  background-color: #ffffff;
`

const formInput = css`
  width: 90%;
  display: flex;
  align-self: center;
  padding: 12px 20px;
  box-sizing: border-box;
  border: 1px solid #f54b3a;
  border-radius: 4px;
  outline: none;
  background-color: #ffffff;
  color: #000000;
  font-size: 1.2em;
  margin-bottom: 1em;
`

const inputStyle = {
  fontSize: '16px',
  outline: 'none',
  border: 'none',
  marginLeft: '0.5em'
}

function FormComponent({ formData, handleSubmit, handleInputChange }) {
  return (
    <form className={cx(formStyle)} onSubmit={(e) => handleSubmit(e)}>
      <div className={cx(formInput)}>
        <label>
          CustomerId:
          <input
            style={inputStyle}
            name="customerId"
            type="text"
            placeholder="CustomerId"
            value={formData.customerID}
            onChange={handleInputChange}
          />
        </label>
      </div>
      <div className={cx(formInput)}>
        <label>
          First Name:
          <input
            style={inputStyle}
            name="first"
            type="text"
            placeholder="First Name"
            value={formData.name.first}
            onChange={handleInputChange}
          />
        </label>
      </div>
      <div className={cx(formInput)}>
        <label>
          Last Name:
          <input
            style={inputStyle}
            name="last"
            type="text"
            placeholder="Last Name"
            value={formData.name.last}
            onChange={handleInputChange}
          />
        </label>
      </div>

      <div className={cx(formInput)}>
        <label>Gender:</label>
        <select
          name="gender"
          style={inputStyle}
          value={formData.gender}
          onChange={handleInputChange}
        >
          <option value="m">Male</option>
          <option value="w">Female</option>
          <option value="o">Other</option>
        </select>
      </div>

      <div className={cx(formInput)}>
        <label>Birthday:</label>
        <div>
          <input
            style={inputStyle}
            name="birthday"
            type="date"
            placeholder="Birthday"
            value={formData.birthday}
            onChange={handleInputChange}
          />
        </div>
      </div>

      <div className={cx(formInput)}>
        <label>Last Contact:</label>
        <div>
          <input
            style={inputStyle}
            name="lastContact"
            type={formData.lastContact.includes('Z') ? 'datetime' : 'datetime-local'}
            placeholder="Last Contact"
            value={
              formData.lastContact.includes('Z')
                ? utcDateConverter(formData.lastContact)
                : formData.lastContact
            }
            onChange={handleInputChange}
          />
        </div>
      </div>

      <div className={cx(formInput)}>
        <label>Lifetime Value:</label>
        <div>
          <input
            style={inputStyle}
            name="customerLifetimeValue"
            type="text"
            value={formData.customerLifetimeValue}
            onChange={handleInputChange}
          />
        </div>
      </div>
    </form>
  )
}

export default FormComponent
