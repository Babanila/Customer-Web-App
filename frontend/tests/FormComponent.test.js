import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import FormComponent from '../src/components/FormComponent'

describe('Check FormComponent Component', () => {
  let sampleFormData, handleSubmit, handleInputChange

  beforeEach(() => {
    handleSubmit = jest.fn()
    handleInputChange = jest.fn()
    sampleFormData = {
      customerID: 7,
      name: {
        first: 'Sarah',
        last: 'Conol'
      },
      birthday: '1986-03-21',
      gender: 'w',
      lastContact: '2017-06-01T23:28:56.782Z',
      customerLifetimeValue: 2191
    }
  })

  it('should renders component without crashing', () => {
    const tree = renderer
      .create(
        <FormComponent
          formData={sampleFormData}
          handleSubmit={handleSubmit}
          handleInputChange={handleInputChange}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correct number of element in FormComponent component', () => {
    const FormComponentWrapper = shallow(
      <FormComponent
        formData={sampleFormData}
        handleSubmit={handleSubmit}
        handleInputChange={handleInputChange}
      />
    )

    const actualDiv = FormComponentWrapper.find('div')
    const actualLabel = FormComponentWrapper.find('label')
    const actualInput = FormComponentWrapper.find('input')
    expect(FormComponent).toBeDefined()
    expect(actualDiv.exists).toBeTruthy()
    expect(actualDiv.length).toEqual(10)
    expect(actualLabel.length).toEqual(7)
    expect(actualInput.length).toEqual(6)
  })
})
