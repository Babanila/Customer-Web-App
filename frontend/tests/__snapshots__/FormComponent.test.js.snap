// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check FormComponent Component should renders component without crashing 1`] = `
<form
  className="css-cgyezj"
  onSubmit={[Function]}
>
  <div
    className="css-ku1ga7"
  >
    <label>
      CustomerId:
      <input
        name="customerID"
        onChange={[MockFunction]}
        placeholder="CustomerId"
        style={
          Object {
            "border": "none",
            "fontSize": "16px",
            "marginLeft": "0.5em",
            "outline": "none",
          }
        }
        type="text"
        value={7}
      />
    </label>
  </div>
  <div
    className="css-ku1ga7"
  >
    <label>
      First Name:
      <input
        name="first"
        onChange={[MockFunction]}
        placeholder="First Name"
        style={
          Object {
            "border": "none",
            "fontSize": "16px",
            "marginLeft": "0.5em",
            "outline": "none",
          }
        }
        type="text"
        value="Sarah"
      />
    </label>
  </div>
  <div
    className="css-ku1ga7"
  >
    <label>
      Last Name:
      <input
        name="last"
        onChange={[MockFunction]}
        placeholder="Last Name"
        style={
          Object {
            "border": "none",
            "fontSize": "16px",
            "marginLeft": "0.5em",
            "outline": "none",
          }
        }
        type="text"
        value="Conol"
      />
    </label>
  </div>
  <div
    className="css-ku1ga7"
  >
    <label>
      Gender:
    </label>
    <select
      name="gender"
      onChange={[MockFunction]}
      style={
        Object {
          "border": "none",
          "fontSize": "16px",
          "marginLeft": "0.5em",
          "outline": "none",
        }
      }
      value="w"
    >
      <option
        value=""
      >
         Choose one
      </option>
      <option
        value="m"
      >
        Male
      </option>
      <option
        value="w"
      >
        Female
      </option>
      <option
        value="o"
      >
        Other
      </option>
    </select>
  </div>
  <div
    className="css-ku1ga7"
  >
    <label>
      Birthday:
    </label>
    <div>
      <input
        name="birthday"
        onChange={[MockFunction]}
        placeholder="Birthday"
        style={
          Object {
            "border": "none",
            "fontSize": "16px",
            "marginLeft": "0.5em",
            "outline": "none",
          }
        }
        type="date"
        value="1986-03-21"
      />
    </div>
  </div>
  <div
    className="css-ku1ga7"
  >
    <label>
      Last Contact:
    </label>
    <div>
      <input
        name="lastContact"
        onChange={[MockFunction]}
        placeholder="Last Contact"
        style={
          Object {
            "border": "none",
            "fontSize": "16px",
            "marginLeft": "0.5em",
            "outline": "none",
          }
        }
        type="datetime"
        value="Thu, 01 Jun 2017 23:28:56 GMT"
      />
    </div>
  </div>
  <div
    className="css-ku1ga7"
  >
    <label>
      Lifetime Value:
    </label>
    <div>
      <input
        name="customerLifetimeValue"
        onChange={[MockFunction]}
        style={
          Object {
            "border": "none",
            "fontSize": "16px",
            "marginLeft": "0.5em",
            "outline": "none",
          }
        }
        type="text"
        value={2191}
      />
    </div>
  </div>
</form>
`;
